### ISSUE TITLE
'react/no-unused-prop-types' not working properly

### ISSUE THREAD

**Issue by yury-dymov** [eslint/eslint#7123]
Situation bellow is considered as an error:

```js
const propTypes = {
  clientVersion:  PropTypes.string,
};

const defaultProps = {
  clientVersion: null,
};

class App extends Component {
  constructor(props) {
    super(props);

    this.checkClientVersion = this.checkClientVersion.bind(this);
  }

  checkClientVersion(props = this.props) {
    const { clientVersion } = props;
   <...>
  }

  render() {
    <...>
}

App.propTypes     = propTypes;
App.defaultProps  = defaultProps;

export default App;
```

### Versions
```js
    "eslint": "3.5.0",
    "eslint-config-airbnb": "11.1.0",
    "eslint-loader": "1.5.0",
    "eslint-plugin-import": "1.14.0",
    "eslint-plugin-jsx-a11y": "2.2.1",
    "eslint-plugin-react": "6.2.0",
```

**Comment by eslintbot** [comment]
Hi @yury-dymov, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Besides the info we've asked for above, I'd like to also ask why you think this is an ESLint issue specifically, rather than eslint-plugin-react. Have you opened an issue on their repository as well?

**Comment by yury-dymov** [comment]
Hello @platinumazure,

 1. I didn't open issue there. I don't know much, how eslint works under the hood, therefore I haven't considered that it might be not your bug. I am sorry for inconveniences then.
 2. I was puzzled by "needs info" response. Eslint version was provided, reproducible source code was provided, affected rule was mentioned in the subject. My expectations are clear: eslint (or plugin) found error in the code telling me that I am not using prop, which I am clearly using.
